### PR TITLE
[SPARK-49000][SQL][FOLLOWUP] Improve code style and update comments

### PR DESCRIPTION
**Pull request by uros-db**
### What changes were proposed in this pull request?
Fix `RewriteDistinctAggregates` rule to deal properly with aggregation on DISTINCT literals. Physical plan for `select count(distinct 1) from t`:

```
-- count(distinct 1)
== Physical Plan ==
AdaptiveSparkPlan isFinalPlan=false
+- HashAggregate(keys=[], functions=[count(distinct 1)], output=[count(DISTINCT 1)#2L])
   +- HashAggregate(keys=[], functions=[partial_count(distinct 1)], output=[count#6L])
      +- HashAggregate(keys=[], functions=[], output=[])
         +- Exchange SinglePartition, ENSURE_REQUIREMENTS, [plan_id=20]
            +- HashAggregate(keys=[], functions=[], output=[])
               +- FileScan parquet spark_catalog.default.t[] Batched: false, DataFilters: [], Format: Parquet, Location: InMemoryFileIndex(1 paths)[file:/Users/nikola.mandic/oss-spark/spark-warehouse/org.apache.spark.s..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<>
```

Problem is happening when `HashAggregate(keys=[], functions=[], output=[])` node yields one row to `partial_count` node, which then captures one row. This four-node structure is constructed by `AggUtils.planAggregateWithOneDistinct`.

To fix the problem, we're adding `Expand` node which will force non-empty grouping expressions in `HashAggregateExec` nodes. This will in turn enable streaming zero rows to parent `partial_count` node, yielding correct final result.

### Why are the changes needed?
Aggregation with DISTINCT literal gives wrong results. For example, when running on empty table `t`:
`select count(distinct 1) from t` returns 1, while the correct result should be 0.
For reference:
`select count(1) from t` returns 0, which is the correct and expected result.

### Does this PR introduce _any_ user-facing change?
Yes, this fixes a critical bug in Spark.

### How was this patch tested?
New e2e SQL tests for aggregates with DISTINCT literals.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by uros-db** [comment]
@cloud-fan @yaooqinn @viirya follow-up for SPARK-49000 is ready

back-ports are in separate PRs

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala [hunk 1]
```diff
@@ -198,13 +198,15 @@ import org.apache.spark.util.collection.Utils
  */
 object RewriteDistinctAggregates extends Rule[LogicalPlan] {
   private def mustRewrite(
-      aggregateExpressions: Seq[AggregateExpression],
+      distinctAggs: Seq[AggregateExpression],
       groupingExpressions: Seq[Expression]): Boolean = {
-    // If there are any AggregateExpressions with filter, we need to rewrite the query.
+    // If there are any distinct AggregateExpressions with filter, we need to rewrite the query.
     // Also, if there are no grouping expressions and all aggregate expressions are foldable,
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala:204]:
```suggestion
    // Also, if there are no grouping expressions and all distinct aggregate expressions are foldable,
```

**Comment by dongjoon-hyun** [comment]
In addition, please revise the PR title into a more corresponding one instead of sharing the same title with the original PR. The current PR has identical one, `[SPARK-49000][SQL][FOLLOWUP] Fix "select count(distinct 1) from t" where t is empty table by expanding RewriteDistinctAggregates`.
```
$ git log --oneline | grep SPARK-49000
dfa21332f20 [SPARK-49000][SQL] Fix "select count(distinct 1) from t" where t is empty table by expanding RewriteDistinctAggregates
```

**Review by cloud-fan — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thank you @uros-db @dongjoon-hyun @cloud-fan 

Merged to master

**Pull request closed without merging**